DB/src/main/java/io/deephaven/db/v2/SourceTableMap.java

```diff
@@ -119,12 +123,11 @@ private void sortAndAddLocations(@NotNull final Stream<TableLocation> locations)
 
     private Table makeTable(@NotNull final TableLocation tableLocation) {
         return applyTablePermissions.apply(new PartitionAwareSourceTable(
-                getConstituentDefinition().orElse(null),
+                getConstituentDefinition().orElseThrow(IllegalStateException::new),
```

Comment:
It literally can't happen, because we supply the (final, not-null) constituent definition in the constructor; I just wanted the IDE to shut up. I feel like this is the appropriate amount of fanciness for what is in effect an assertion that reality is real.